### PR TITLE
change image analysis queue processing behavior from first come first served to fair share across account

### PR DESCRIPTION
**Pull request by nurmi**
Implements #491

### CODE REVIEW

**Comment by KnoxAnderson** [comment]
Have you thought about prioritizing images coming from the inline scanner at the top of the list as well?

Ie images submitted with an analyze in the inline scan get prioritized above a standard image add submission

**Comment by nurmi** [comment]
> Have you thought about prioritizing images coming from the inline scanner at the top of the list as well?
> 
> Ie images submitted with an analyze in the inline scan get prioritized above a standard image add submission

Hi @KnoxAnderson - we're going to be tracking the ability for a user to explicitly 'prioritize' the analysis of an image separately from adjusting the default queue task processing behavior.  This would extend to any image analysis 'add' operation ideally, and may be part of a bigger enhancement related to task queue management - not on the roadmap yet but it is on the list of enhancements for future consideration!

File: anchore_engine/services/catalog/__init__.py [hunk 1]
```diff
@@ -979,22 +985,39 @@ def handle_analyzer_queue(*args, **kwargs):
...
+                                    queue_rebalance[userId]['lowest_queueId'] = q_record.get('queueId')
+                                if q_record.get('queueId') < 0 and q_record.get('queueId') >= highest_neg_queueId:
+                                    highest_neg_queueId = q_record.get('queueId')
+                            except Exception as err:
+                                logger.error("failed to store image current queueID - excpetion: {}".format(err))
+                                
                     except Exception as err:
                         logger.error("failed to check/queue image for analysis - exception: " + str(err))
 
+    # promote queued tasks into the analysis queue such that one image from each account is prioritized, to implement a simple 'fair share' across accounts
```
Comment (zhill) [inline anchore_engine/services/catalog/__init__.py:1010]: I'd like to see this split out into its own function with some test coverage of the behavior so we can catch any regressions later.

Comment (nurmi) [reply anchore_engine/services/catalog/__init__.py:1010]: the logic has been adjusted to support testability, and a unit test has been added

**Pull request merged**